userspace/falco/app/actions/init_inspectors.cpp

```diff
@@ -224,17 +227,18 @@ falco::app::run_result falco::app::actions::init_inspectors(falco::app::state& s
 		// in live mode, each inspector should have registered at most two event sources:
 		// the "syscall" on, loaded at default at index 0, and optionally another
 		// one defined by a plugin, at index 1
-		if(!s.is_capture_mode()) {
-			const auto& sources = src_info->inspector->event_sources();
-			if(sources.size() == 0 || sources.size() > 2 ||
-			   sources[0] != falco_common::syscall_source) {
-				err.clear();
-				for(const auto& source : sources) {
-					err += (err.empty() ? "" : ", ") + source;
-				}
-				return run_result::fatal("Illegal sources setup in live inspector for source '" +
-				                         src + "': " + err);
+		if(is_capture_mode) {
```

Comment:
Shouldn't the comment just above the if be moved below close the source size check ?
It describe the logic in the body quite well however reading it and then having the capture mode check and continue statement makes things harder to reason about.

Reply:
Yeah you are right. I was not sure which of the two styles to prefer but I guess your alternative is better. Gonna fix it here and in other places where the same logic applies.